### ISSUE TITLE
[question] Fixed vs Dynamic List Size

### ISSUE THREAD

**Issue by hdatteln** [tensorflow/ranking#121]
Hi, 
I am wondering about the list_size parameter for `example_serving_input_fn` and `build_ranking_dataset`.

My training data would have lists of variable sizes, so i think i need to set the list_size to 'None' (i.e. dynamic) both during training and when inferring predictions.
Or would it be better to set the list_size to the maximum list length during training? Just wondering if it would make a difference

**Comment by xuanhuiwang** [comment]
They both should be the same in terms of modeling. 

Setting list_size to None will be more efficient. If list_size is set to maximum list length, all batches will have the shape [batch_size, maximum_list_length, ...]. When it is None, the 2nd dim in each batch will be the maximum list length in that batch, instead of the maximum one in the whole data set.

**Comment by hdatteln** [comment]
Great, thanks for clarifying!
